### PR TITLE
Add TAPscan tool

### PR DESCRIPTION
**Pull request by shiltemann**
Tool to detect Transcript Associated Proteins (TAPs) 

ping @bgruening. This tool allows you to search for TAPs in your own sequences. I will create a second tool later which will be more of a data-source type tool, where you can download known TAPs from the TAP database, similar to what you can do on the TAPscan website (https://plantcode.cup.uni-freiburg.de/tapscan). 

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

**Comment by shiltemann** [comment]
Ping also @deeptivarshney, in case you want to review, or maybe have some more useful information to add to the help text?

File: tools/tapscan/tapscan.xml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+hmmsearch
+  --domtblout domtblout.txt
+  --cut_ga
+  '${__tool_directory__}/tapscan_domains_v12.txt'
+  '$protein_fasta_in'
+
+&&
+
+perl '${__tool_directory__}/tapscan_script_v74.pl'
```
Comment (bgruening) [inline tools/tapscan/tapscan.xml:17]: Can we add those files as required_files? https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-required-files

File: tools/tapscan/tapscan.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
+<tool id="tapscan_classify" name="TAPScan Classify" version="4.74+galaxy0">
+    <description>Detect Transcription Associated Proteins (TAPs)</description>
+    <requirements>
+        <requirement type="package" version="3.3.2">hmmer</requirement>
+        <requirement type="package" version="5.26">perl</requirement>
```
Comment (bgruening) [inline tools/tapscan/tapscan.xml:5]: We should add sed here, if we make excessive use of it.

File: tools/tapscan/tapscan.xml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+sed -i '1s/^/TAP family\tnumber of detected proteins\n/' '$taps_family_counts' &&
+sed -i '1s/^/sequence ID\tTAP family\tSubfamily\tnumber of classifications\tdomains\n/' '$taps_detected_extra'
+
+    ]]></command>
+    <inputs>
+        <param name="protein_fasta_in" type="data" format="fasta" optional="false" label="Proteins in FASTA format" help=""/>
+        <param name="domtblout" type="boolean" checked="false" label="Output the HMMer domain hits table?"/>
+    </inputs>
+    <outputs>
+        <data name="taps_detected" format="tabular" label="${tool.name} on ${on_string}: Detected TAPs - domains and assigned TAP family
```
Comment (bgruening) [inline tools/tapscan/tapscan.xml:43]: Can you please add the column names via metadata here: https://github.com/galaxyproject/tools-iuc/blob/3421a08ab29b58cd514011767f429f6769e3e86d/tools/onto_toolkit/get_terms.xml#L17

File: tools/tapscan/tapscan.xml [hunk 4]
```diff
@@ -0,0 +1,101 @@
+<tool id="tapscan_classify" name="TAPScan Classify" version="4.74+galaxy0">
```
[the remark this answers is not in the document]
Comment (shiltemann) [reply tools/tapscan/tapscan.xml]: I went with 23.0, let me know if it needs to be something different

**Comment by shiltemann** [comment]
ok, can't get the domains file under the 1MB file size limit, with compression it's still 1.7 MB, so will move this tool to https://github.com/bgruening/galaxytools/

**Pull request closed without merging**